Error when decrypting.

Hi,

I'm getting the following error when decrypting:
```
gauth
Encryption password:
panic: runtime error: index out of range

goroutine 1 [running]:
main.main()
	/Users/warren/Go/src/github.com/pcarrier/gauth/gauth.go:106 +0x8d8

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1696 +0x1

goroutine 20 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 34 [select, locked to thread]:
runtime.gopark(0x1a2390, 0xc8200bc728, 0x164fd0, 0x6, 0x2c818, 0x2)
	/usr/local/go/src/runtime/proc.go:185 +0x163
runtime.selectgoImpl(0xc8200bc728, 0x0, 0x18)
	/usr/local/go/src/runtime/select.go:392 +0xa64
runtime.selectgo(0xc8200bc728)
	/usr/local/go/src/runtime/select.go:212 +0x12
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal1_unix.go:227 +0x323
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1696 +0x1
```

I'm using the following versions:
```
go version
go version go1.5.1 darwin/amd64
openssl version
OpenSSL 0.9.8zg 14 July 2015
sw_vers
ProductName:	Mac OS X
ProductVersion:	10.10.5
BuildVersion:	14F1021
```

Any thoughts?

--- a/gauth.go
+++ b/gauth.go
@@ -2,7 +2,6 @@ package main
 
 import (
 	"bytes"
-	"code.google.com/p/gopass"
 	"crypto/aes"
 	"crypto/cipher"
 	"crypto/hmac"
@@ -11,12 +10,14 @@ import (
 	"encoding/base32"
 	"encoding/csv"
 	"fmt"
+	"golang.org/x/crypto/ssh/terminal"
 	"io/ioutil"
 	"log"
 	"math/big"
 	"os/user"
 	"path"
 	"strings"
+	"syscall"
 	"time"
 )
 
@@ -82,7 +83,8 @@ func main() {
 
 	// Support for 'openssl enc -aes-128-cbc -md sha256 -pass pass:'
 	if bytes.Compare(cfgContent[:8], []byte{0x53, 0x61, 0x6c, 0x74, 0x65, 0x64, 0x5f, 0x5f}) == 0 {
-		passwd, e := gopass.GetPass("Encryption password: ")
+		fmt.Printf("Encryption password: ")
+		passwd, e := terminal.ReadPassword(syscall.Stdin)
 		if e != nil {
 			log.Fatal(e)
 		}
@@ -102,7 +104,7 @@ func main() {
 		mode := cipher.NewCBCDecrypter(block, iv)
 		mode.CryptBlocks(rest, rest)
 		// Remove padding
-		i := len(rest)
+		i := len(rest) - 1
 		for rest[i] < 16 {
 			i--
 		}

